includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2617,12 +2589,12 @@ public function create_and_confirm_setup_intent() {
 			$customer_id   = $this->customer_service->create_customer_for_user( $user, $customer_data );
 		}
 
-		return $this->payments_api_client->create_and_confirm_setup_intent(
-			$payment_information->get_payment_method(),
-			$customer_id,
-			$should_save_in_platform_account,
-			$this->is_platform_payment_method( $payment_information->is_using_saved_payment_method() )
-		);
+		$create_and_confirm_intent_request = Create_And_Confirm_Setup_Intention::create();
+		$create_and_confirm_intent_request->set_customer( $customer_id );
+		$create_and_confirm_intent_request->set_payment_method( $payment_information->get_payment_method() );
+		$intent = $create_and_confirm_intent_request->send( 'wcpay_create_and_confirm_setup_intention_request', $payment_information, $should_save_in_platform_account, false );
+
+		return $intent->to_array();
```

Comment:
Is there something, which requires the intent to be an actual array? Unless there is a strong `array` typehint in a function, this conversion should not be needed.

Reply:
I returned default Intent or Request object now as you suggested. I still used `->to_array()` when rest response is returned. The reason behind this is that I want to be sure that correct data or array will be used.